tests/integration/azure.py

```diff
@@ -606,6 +612,63 @@ def upload_image(self, progress_function = None):
             ).result()
 
             self.logger.info(f"Image {image_file} uploaded as {image_name}")
+
+            gallery_name = self.config['gallery_name']
+            gallery_image_name = self.config['gallery_image_name']
+            gallery_image_version_name = self.config['gallery_image_version_name']
+            self.cclient.galleries.begin_create_or_update(
+                resource_group_name = self._resourcegroup.name,
+                gallery_name = gallery_name,
+                gallery = {
+                    'location': self._resourcegroup.location,
+                }
+            )
+
+            allowed_architectures = ["x64", "Arm64"]
```

Comment:
Please have these checks very early in the flow of the tests (preferably in the init method or in `validate_config`) as it would make the test fail late because of invalid input and leave orphaned resources behind.